backend/app2/services/file_service.py

```
@@ -0,0 +1,340 @@
+                    return file
+                    
+            except NotFoundException:
+                # If file not found directly, try finding through tracks
+                if user_id:
+                    logger.info(f"File {file_id} not found directly, trying to find through tracks")
+                    
+                    tracks = await self.track_repository.get_by_file_id(file_id, file_type)
+                    
+                    if not tracks:
+                        logger.error(f"{file_type.capitalize()} file {file_id} not found")
+                        raise NotFoundException(f"{file_type.capitalize()} file", file_id)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)